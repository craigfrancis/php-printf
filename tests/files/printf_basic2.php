<?php
echo "*** Testing printf() : basic functionality - using integer format ***\n";


// Initialise all required variables
$format = "format";
$format1 = "%d";
$format2 = "%d %d";
$format3 = "%d %d %d";
$arg1 = 111;
$arg2 = 222;
$arg3 = 333;

echo "\n-- Calling printf() with no arguments --\n";
$result = user_printf($format);
echo "\n";
var_dump($result);

echo "\n-- Calling printf() with one arguments--\n";
$result = user_printf($format1, $arg1);
echo "\n";
var_dump($result);

echo "\n-- Calling printf() with two arguments--\n";
$result = user_printf($format2, $arg1, $arg2);
echo "\n";
var_dump($result);

echo "\n-- Calling printf() with three arguments--\n";
$result = user_printf($format3, $arg1, $arg2, $arg3);
echo "\n";
var_dump($result);

?>
