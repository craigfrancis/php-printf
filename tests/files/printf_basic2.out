*** Testing printf() : basic functionality - using integer format ***

-- Calling printf() with no arguments --

Warning: Undefined variable $args in /Volumes/WebServer/Projects/php.fprint/functions.php on line 29

Fatal error: Uncaught TypeError: user_vprintf(): Argument #2 ($values) must be of type array, null given, called in /Volumes/WebServer/Projects/php.fprint/functions.php on line 29 and defined in /Volumes/WebServer/Projects/php.fprint/functions.php:22
Stack trace:
#0 /Volumes/WebServer/Projects/php.fprint/functions.php(29): user_vprintf('format', NULL)
#1 /Volumes/WebServer/Projects/php.fprint/tests/files/printf_basic2.php(15): user_printf('format')
#2 {main}
  thrown in /Volumes/WebServer/Projects/php.fprint/functions.php on line 22